[ivyy-poison] iP

ivyy-poison opened the pull request:
### Eduk Chatbot

> “My mind kinda empty lol" – Ivan Tan ([source](https://github.com/ivyy-poison))

Eduk frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉😄

And it is **FREE**!

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)
- [ ]  Help me pass this mod (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:


```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

ivyy-poison committed:
Achieve Level 0

ivyy-poison committed:
Achieve Level 1

ivyy-poison committed:
Achieve Level 2

ivyy-poison committed:
Achieve Level 3

ivyy-poison committed:
Achieve Level 4

ivyy-poison committed:
Achieve Text UI Testing

ivyy-poison committed:
Achieve Level 5

ivyy-poison committed:
Achieve Level 6

ivyy-poison committed:
Achieve A-Enums

ivyy-poison committed:
Refactor code to better adhere to standards

* Remove unnecessary indentations on line breaks and switch block

ivyy-poison committed:
Implement basic read and write into data file

* Read from text file to generate event list
* Write events into text file using chatbot interface

ivyy-poison committed:
Implement save into storage for mark and delete function

* Update gitignore to exclude /data folder in commits to github
* Implement overrideStorage static method to update data file after each marked or deleted event

ivyy-poison committed:
Incorporate date-time format into implementation

* Store datetimes as LocalDateTime objects in Tasks
* Accept only well-formatted date times as valid inputs to create new tasks
* Display and render tasks with differently formatted date time

ivyy-poison committed:
Implement classes for Parser, Ui, Storage, and TaskList

* Refactor code to better adhere to OOP design principles
* Make minor tweaks to formatting of various event types for bugfix
* Create Parser, Ui, Storage and TaskList classes

ivyy-poison committed:
Refactor code and Achieve a-MoreOOP

ivyy-poison committed:
Organize classes into packages

ivyy-poison committed:
Initialise Gradle file

ivyy-poison committed:
Merge branch 'add-gradle-support'

* Merge add-gradle-support branch into master in order to incorporate gradle support into Duke chatbot

ivyy-poison committed:
Merge add-gradle-support to integrate gradle with packaging

Merge branch 'add-gradle-support' into a-Packages

ivyy-poison committed:
Untrack data folder

ivyy-poison committed:
Add gradle support to ip project

* Refactor code and arrange into file directories supported by gradle
* Tweak build.gradle file and add main Java class.

ivyy-poison committed:
Merge commit 'e1590948' into main

ivyy-poison committed:
Implement JUnit tests for ToDo and Deadline classes

ivyy-poison committed:
Tweak build.gradle and create JAR file

ivyy-poison committed:
Add JavaDoc comments for all public methods in Duke package

ivyy-poison committed:
Implement search feature for tasks in task list

* Implement new filter method in TaskList that returns a new TaskList object
* Implement new method to handle find command

ivyy-poison committed:
Refactor code to improve coding standard

* Reorder package imports to improve readability in determining dependencies

ivyy-poison committed:
Merge branch 'branch-A-Javadoc'

ivyy-poison committed:
Merge branch 'branch-Level-9'

ivyy-poison committed:
Implement tutorial part 1

ivyy-poison committed:
Implement tutorial part 2

ivyy-poison committed:
Implement tutorial part 3

ivyy-poison committed:
Implement tutorial part 4

ivyy-poison committed:
Integrate chatbot functionalities with JavaFX UI

ivyy-poison committed:
Add checkstyle checks to gradle project

ivyy-poison committed:
Refactor code to adhere to style rules

* Ran checkstyle on code in main folder and fixed most errors

ivyy-poison committed:
Merge branch 'branch-Level-10'

jinyuan0425 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,207 @@
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
```
Perhaps you can update the header comment to reflect the description of the method better.

ivyy-poison replied:
Right, thanks for the suggestion, looks like my code needs some cleanup lol

jinyuan0425 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,207 @@
+        ekud.run();
+    }
+
+    // private void handleCommand(CommandType commandType, String command) throws DukeException {
+    //     switch (commandType) {
+    //     case LIST:
+    //         ui.printList(tasks.getTasks());
+    //         break;
+    //     case MARK:
+    //         markTask(command);
+    //         break;
+    //     case DELETE:
+    //         deleteTask(command);
+    //         break;
+    //     case TODO:
+    //     case DEADLINE:
+    //     case EVENT:
+    //         addTask(command);
+    //         break;
+    //     case FIND:
+    //         handleFind(command);
+    //         break;
+    //     case UNKNOWN:
+    //         ui.printErrorMessage(new DukeException("I'm sorry, but I don't know what that means :-("));
+    //         break;
+    //     }
+    // }
```
Mayhap can consider removing commented out code to improve readability.

ivyy-poison replied:
Right, thank you!

jinyuan0425 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,207 @@
+            task = deleteTask(command);
+            return ui.displayDeletedTaskConfirmation(task, tasks);
+
+        case TODO:
+        case DEADLINE:
```
Should comment // Fallthrough for TODO and DEADLINE cases.

ivyy-poison replied:
This was an intentional choice, as EVENT, TODO and DEADLINE runs through the same event handler addTask(command)

jinyuan0425 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,115 @@
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+
+
+
+
+
+
```
Can consider removing the whitespace.

ivyy-poison replied:
Thank you!

jinyuan0425 reviewed (commented):
Generally well done, code is highly readable!

Perhaps can clean up the code slightly by removing parts such as commented out code and excessive whitespace.

ivyy-poison committed:
Refactor code to better adhere to coding standard

Currently, the code is functional, but work can be done to make
the code more readable, scalable and maintainable

* Remove blank, empty lines
* Refactor Javadoc comments to be more concise, yet informative

By doing the above, we make the code more readable. In addition,
an informative Javadoc comment helps other developers better
understand the purpose of a method or class.

ivyy-poison committed:
Add assertions to document important assumptions

Assertions are important in documenting important assumptions that
should hold at various points in the code

* Add assertion statements at the end of various methods

These methods mostly consist of command handlers and a handleCommand
method. Proper exception handling has been put in place to take care
of invalid responses, therefore the code should never reach the
assertion, and return null or empty string.

ivyy-poison committed:
Remove dead code

WeeMingQing commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,207 @@
+            ui.printErrorMessage(e);
+        }
+        return null;
+    }
```
Good job in avoiding deep nesting methods here and in other places as well.

WeeMingQing commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,107 @@
+     * @throws DukeException If the given index is invalid.
+     */
+    public Task get(int index) throws DukeException {
+        if (index < 0 || index >= tasks.size()) {
```
Well Done in avoiding complicated expressions.

WeeMingQing reviewed (commented):
LGTM.

ivyy-poison committed:
Add gradle.yml file to integrate CI into Github Actions

ivyy-poison committed:
Add assertion statements to more classes

Assertion statements are used to document important assumptions
that should hold at various points in the code.

Assertion statements were added to the following classes:
* Storage.java
* TaskList.java
* Parser.java
* Ui.java

ivyy-poison committed:
Change build.gradle file to enable assertions on run

ivyy-poison committed:
Refactor code to better adhere to stylecheck

sunzihan23 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,207 @@
+
+    /**
+     * Main entry point of the Duke application.
+     * @param args Command line arguments.
```
Would be good to have an empty line between description and parameter sections! But looks like other JavaDoc comments have empty lines so this is not major issue

sunzihan23 commented on src/main/java/duke/components/DialogBox.java:
```diff
@@ -0,0 +1,64 @@
+    @FXML
+    private Label dialog;
+    // @FXML
+    // private ImageView displayPicture;
```
Maybe these can be removed if they're not in use?

ivyy-poison committed:
Merge pull request #2 from ivyy-poison/branch-A-CodeQuality

Refactor code to improve code quality

ivyy-poison committed:
Merge branch 'master' into branch-A-Assertions

ivyy-poison committed:
Merge pull request #3 from ivyy-poison/branch-A-Assertions

Add assertions to code

ivyy-poison committed:
Merge branch 'master' into branch-A-CI

ivyy-poison committed:
Refactor code to meet checkstyle requirements

ivyy-poison committed:
Implement command to view schedule across time period

* Create new temporal interface for Deadline and Event tasks to implement
* Extend parser to handle view schedule command
* Implement filterByDateRange method in TaskList to filter by date range

ivyy-poison committed:
Refactor code to fulfill stylecheck requirements

ivyy-poison committed:
Fix bug in DeadlineTest.java

ivyy-poison committed:
Improve GUI of Duke chatbot

ivyy-poison committed:
Add brief user guide and screenshot

ivyy-poison committed:
Tweak UI to display message when tasklist is empty

ivyy-poison committed:
Implement chatbot welcome message

ivyy-poison committed:
Implement error handling for mark and delete task

ivyy-poison committed:
Implement more unit tests for Deadline and Todo

ivyy-poison committed:
Improve Readme
